src/client/Microsoft.Identity.Client.Broker/BrokerExtension.cs

```diff
@@ -47,29 +47,55 @@ public static PublicClientApplicationBuilder WithBrokerPreview(this PublicClient
         /// parameters, and to create a public client application instance</returns>
         public static PublicClientApplicationBuilder WithBroker(this PublicClientApplicationBuilder builder, BrokerOptions brokerOptions)
         {
-            AddRuntimeSupportForWam(builder);
+            AddRuntimeSupport(builder, true);
             builder.Config.BrokerOptions = brokerOptions;
             builder.Config.IsBrokerEnabled = brokerOptions.IsBrokerEnabledOnCurrentOs();
             return builder;
         }
 
-        private static void AddRuntimeSupportForWam(PublicClientApplicationBuilder builder)
+        /// <summary>
+        /// Use this API to enable SsoPolicy enforcement when not using broker.
+        /// 
+        /// </summary>
+        /// <param name="builder"></param>
+        /// <returns>A <see cref="PublicClientApplicationBuilder"/> from which to set more
+        /// parameters, and to create a public client application instance</returns>
+        public static PublicClientApplicationBuilder WithSsoPolicy(this PublicClientApplicationBuilder builder)
+        {
+            AddRuntimeSupport(builder, false);
+            builder.Config.IsSsoPolicyEnabled = true;
+            return builder;
+        }
+
+        private static void AddRuntimeSupport(PublicClientApplicationBuilder builder, bool isWamBrokerScenario)
         {
             if (DesktopOsHelper.IsWin10OrServerEquivalent())
             {
-                builder.Config.BrokerCreatorFunc =
-                     (uiParent, appConfig, logger) =>
-                     {
-                         logger.Info("[RuntimeBroker] WAM supported OS.");
-                         return new RuntimeBroker(uiParent, appConfig, logger);
-                     };
+                if (isWamBrokerScenario)
+                {
+                    builder.Config.BrokerCreatorFunc =
+                         (uiParent, appConfig, logger) =>
+                         {
+                             logger.Info("[Runtime] WAM supported OS.");
+                             return new RuntimeBroker(uiParent, appConfig, logger);
+                         };
+                }
+                else
+                {
+                    builder.Config.BrokerCreatorFunc =
+                         (uiParent, appConfig, logger) =>
+                         {
+                             logger.Info("[Runtime] SsoPolicy supported OS.");
+                             return new RuntimeBroker(appConfig, logger);
```

Comment:
Why did you need to create a new ctor? The ctor does not validate the widown handle (uiParent), only `broker.AcquireTokenInteractive` does.

Reply:
I actually realized later this was  not needed when I created the broker instance.  I will remove this.